### ISSUE TITLE
CPU is maxed out after adding five repositories

### ISSUE THREAD

**Issue by dnoland1** [rancher/fleet#185]
Steps to reproduce:
1. Set up Rancher v2.5.3. Using 2 m5a.large (2vCPU, 8GB RAM) on AWS with k3s and RDS/Aurora/MySQL.
2. Launch 5 VMs to be used for downstream clusters. Using t3a.medium instance on AWS.
3. Create 5 custom clusters, test1, test2, test3, test4, test5.
4. Create one node etcd/cp/worker clusters using the VMs created in step 2.
5. Go to Continuous Delivery
6. Create a new Fleet repository. I'm using https://github.com/chrisurwin/fleet-demo.git and main branch. Name it fleet1 and only add test1 cluster.
7. Wait about 10-15 minutes for the workloads to deploy.
8. Repeat step 6-7 four times, but use fleet2,3,4,5 for the repo names and test2,3,4,5 for the clusters.

Results:
After adding each fleet repository, the fleet-controller is consuming about 20% of the CPU. After adding all 5 repos, 100% CPU is being used. Here's the CPU graph:
![image](https://user-images.githubusercontent.com/468462/102383747-bfbb0280-3f80-11eb-9eb3-da5437a29411.png)

repos were added at approximately 7:20, 7:37, 7:52, 8:06, and 8:21
Here's a top sample:
```
top - 16:28:57 up  1:38,  0 users,  load average: 3.21, 3.37, 3.05
Tasks: 145 total,   2 running, 141 sleeping,   0 stopped,   2 zombie
%Cpu(s): 97.3 us,  2.0 sy,  0.0 ni,  0.5 id,  0.0 wa,  0.0 hi,  0.2 si,  0.0 st
MiB Mem :   7710.5 total,    162.0 free,   2955.6 used,   4592.9 buff/cache
MiB Swap:      0.0 total,      0.0 free,      0.0 used.   5757.0 avail Mem

    PID USER      PR  NI    VIRT    RES    SHR S  %CPU  %MEM     TIME+ COMMAND
   7438 root      20   0  974180 311956  38492 S 181.1   4.0  77:34.25 fleetcontroller
    940 root      20   0 1874060   1.6g  89568 S  13.0  21.9  20:09.02 k3s-server
```
According to https://rancher.com/docs/rancher/v2.x/en/installation/requirements/#cpu-and-memory a 2vCPU and 8GB RAM  should support up to 150 clusters and 1500 nodes. So only being able to support 5 or less Fleet repositories seems way too low.
gz#14005

**Comment by nikkelma** [comment]
Seeing this also in a prospect's `docker run` installation of 2.5.3, a 4vCPU + 16GB server is being fully utilized by the fleet controller.

**Comment by jiaqiluo** [comment]
The bug is reproduced on a v2.5.4-rc8 HA setup.  The local cluster has 3 nodes,4 CPU, 16 GB of memory for each node. 

The steps are the same as the issue:
- add multiple single-node clusters with ec2 t3a.medium instance
- enable cluster monitoring v1 on the local cluster 
- Create a new Fleet repository using the repo `chrisurwin/fleet-demo.git` and `main` branch. One repo for one cluster 
- wait between adding each repo for the CPU usage to be stable 

Result:
Here is the CPU utilization on the node where fleet-controller runs
<img width="450" alt="Screen Shot 2020-12-30 at 3 01 11 PM" src="https://user-images.githubusercontent.com/6218999/103385930-c9cd1d00-4ab9-11eb-92c9-2981c70050c5.png">

There is a spike in the graph each time a git repo is added, then it goes flat and keeps at a higher value than before adding that git repo.

**Comment by jiaqiluo** [comment]
The bug fix is validated on the v2.5-head HA setup. The local cluster has 3 nodes, 2 CPU, 4 GB of memory for each node.

The steps are the same as the issue:

- add multiple single-node clusters with ec2 t3a.medium instance
- enable cluster monitoring v1 on the local cluster
- Create a new Fleet repository using the repo chrisurwin/fleet-demo.git and main branch. One repo for one cluster
- wait between adding each repo for the CPU usage to be stable


Results:
On the node where fleet-controller runs, the overall cpu usage keeps around 21%.
There is a spike in the graph each time a git repo is added, then the usage reduces to the same level as before adding the git repo.

<img width="407" alt="Screen Shot 2020-12-30 at 9 01 34 PM" src="https://user-images.githubusercontent.com/6218999/103393963-f34e6e80-4ae2-11eb-959d-f9732db54481.png">

(the high usage at the beginning of the timeline is not related to fleet)
